test/parser/parser_test.cpp

```diff
@@ -184,19 +184,19 @@ TEST_F(ParserTestBase, CreateIndexTest) {
   EXPECT_EQ(create_stmt->GetIndexName(), "idx_order");
   EXPECT_EQ(create_stmt->GetTableName(), "oorder");
   EXPECT_EQ(create_stmt->GetIndexAttributes().size(), 2);
-  auto ia1 = create_stmt->GetIndexAttributes()[0].GetExpression();
+  auto ia1 = create_stmt->GetIndexAttributes()[0]->GetExpression();
   EXPECT_EQ(ia1->GetExpressionType(), ExpressionType::OPERATOR_MINUS);
-  auto ia1l = reinterpret_cast<TupleValueExpression *>(ia1->GetChild(0).get());
+  auto ia1l = ia1->GetChild(0).CastManagedPointerTo<const TupleValueExpression>();
```

Comment:
I'll concede that `CastManagedPointerTo` is necessary/useful for the parser test, but I'm not convinced its needed in production code.  Should we make it a private function with this test as a "TEST_FRIEND"?